operator/pkg/manifest/shared.go

```diff
@@ -523,20 +529,24 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 		p = strings.TrimPrefix(p, "spec.")
 		inc, _, err := tpath.GetPathContext(iop, util.PathFromString("spec."+p), true)
 		if err != nil {
-			return "", err
+			return "", "", err
 		}
 		// input value type is always string, transform it to correct type before setting.
 		if err := tpath.WritePathContext(inc, util.ParseValue(v), false); err != nil {
-			return "", err
+			return "", "", err
 		}
 	}
 
 	out, err := yaml.Marshal(iop)
 	if err != nil {
-		return "", err
+		return "", "", err
+	}
+	manifests := strings.Split(iopYAML, "\n---")
+	if len(manifests) > 1 {
+		return string(out), "Multiple documents in one manifest file. First document assumed, the rest is ignored.", nil
```

Comment:
Its valid and common to have a leading or trailing `---` in the file, but only have a single IOP, which may lead to unexpected warnings here

Reply:
I just tested it and after adding mjog@'s suggestion of checking `\n---`, I am not getting a warning with this file:
```
$ cat operator/singleiop.yaml
---
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    enableTracing: false
```
So leading `---` does not warn. Are you sure you don't want to warn for a trailing `---`?
It looks like this:
```
WARNING: Multiple documents in one manifest file. First document assumed, the rest is ignored.
* Overlay file received:
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    enableTracing: false
---

* Overlay file assumed:
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    enableTracing: false
```    
It looks clear that it is safe to ignore the warning but at the same time it educates people that they are not supposed to add anything extra to the file. WDYT?
I can otherwise just check if the `---` is in the end if you prefer.

Reply:
I am not sure I agree with "they are not supposed to add anything extra to the file". Its pretty common from what I have seen. Adding warnings that are not actually an issue can lead to noise + ignoring them